### PR TITLE
release: v1.22.0

### PR DESCRIPTION
**Pull request by bnjjj**
> **Note**
> **This particular PR should be true-merged to `main`.**

This PR represents the merge to `main` of the v1.22.0 release.

This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.

**However!** Some things to review on this PR:

- Does this PR target the right branch? (usually, `main`)
- Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"

If those things look good, this PR is good to merge.

### CODE REVIEW

**Review by o0Ignition0o — APPROVED** [review]

File: apollo-router/src/protocols/multipart.rs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use futures::stream;
+    use serde_json_bytes::ByteString;
+
+    use super::*;
+
+    // TODO add test with empty stream
```
Comment (abernix) [inline apollo-router/src/protocols/multipart.rs:149]: Do we have a ticket for follow-up here?

File: apollo-router/src/services/subgraph.rs [hunk 2]
```diff
@@ -210,3 +235,64 @@ impl Response {
...
+            hasher.update(query.as_bytes());
+        }
+        for (var_name, var_value) in &body.variables {
+            hasher.update(var_name.inner());
+            // TODO implement to_bytes() for value in serde_json_bytes
+            hasher.update(var_value.to_string().as_bytes());
+        }
+        for (name, val) in &body.extensions {
+            hasher.update(name.inner());
+            // TODO implement to_bytes() for value in serde_json_bytes
```
Comment (abernix) [inline apollo-router/src/services/subgraph.rs:292]: Tickets?

**Review by abernix — APPROVED** [review]

File: apollo-router/src/services/subgraph.rs (hunk 2 above)
Comment (bnjjj) [reply apollo-router/src/services/subgraph.rs:292]: https://github.com/apollographql/serde_json_bytes/issues/2 created

File: apollo-router/src/protocols/multipart.rs (hunk 1 above)
Comment (bnjjj) [reply apollo-router/src/protocols/multipart.rs:149]: https://github.com/apollographql/router/issues/3293
